### PR TITLE
feat: Add properties and chainId on build-spec command

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

1. Resolve the [properties and chain type issue](https://github.com/ChainSafe/gossamer/issues/1257) when the **build-spec** command does not parse these values correctly 

2. Includes the [feature to persist a new file using a --output ](https://github.com/ChainSafe/gossamer/issues/1029).

```
before:
gossamer build-spec --raw --genesis-spec ./my-genesis-spec.json > file.json

actual:
gossamer build-spec --raw --genesis-spec ./my-genesis-spec.json --output ./myoutput.json
```

_The first behavior was not removed_

If the file that was passed to **--output** flag exists then an error is returned: **"_file ... already exists, rename to avoid overwritten_"**. 

## Tests

```
1. I change the tests cases **TestBuildFromGenesis** to cover the new parsed fields
2. Created **TestWriteGenesisSpecFile** test case to test the creation of an output file and if the data stored is correct
3. Created **TestWriteGenesisSpecFileWhenFileAlreadyExists** test case to test if the **WriteGenesisSpecFile** returns an error when a filepath already exists
```

## Issues

closes https://github.com/ChainSafe/gossamer/issues/1029
closes https://github.com/ChainSafe/gossamer/issues/1257
closes https://github.com/ChainSafe/gossamer/issues/1030

### CODE REVIEW

File: lib/genesis/helpers.go [hunk 1]
```diff
@@ -141,6 +141,7 @@ func NewGenesisSpecFromJSON(file string) (*Genesis, error) {
 	g := new(Genesis)
 
 	err = json.Unmarshal(data, g)
+
```
Comment (arijitAD) [inline lib/genesis/helpers.go:144]: Remove this.

**Review by arijitAD — COMMENTED** [review]
Looks good. Unit tests seem to be failing. Also, can you generate the spec and verify that it's working with gossamer?

**Comment by EclesioMeloJunior** [comment]
> Looks good. Unit tests seem to be failing. Also, can you generate the spec and verify that it's working with gossamer?

The file was generated with success! But I notice that if the output path file `--output ~/path/to/output/output-json.json`  does not exist the command exit without error but doesn't create the file, so I changed the implementation to create the directory before the file creation if it doesn't exist. In the case that the dir exists then nothing happens.

image

**Review by noot — APPROVED** [review]
it works for me! great work :)

File: dot/build_spec.go [hunk 2]
```diff
@@ -71,6 +78,21 @@ func BuildFromGenesis(path string, authCount int) (*BuildSpec, error) {
 	return bs, nil
 }
 
+// WriteGenesisSpecFile writes the build-spec in the output filepath
+func WriteGenesisSpecFile(data []byte, fp string) error {
+	// if file already exists then dont apply any written on it
+	if utils.PathExists(fp) {
+		return fmt.Errorf("file %s already exists, rename to avoid overwritten", fp)
```
Comment (noot) [inline dot/build_spec.go:85]: change to "rename to avoid overwriting"

**Review by arijitAD — APPROVED** [review]

**Pull request merged**